feat: add `stats/nanmax`

gururaj1512 opened the pull request:
Resolves None

## Description

> What is the purpose of this pull request?

This pull request:

-   feat: add `stats/nanmax`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

gururaj1512 committed:
feat: add `stats/nanmax`

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into stats-nanmax

kgryte commented on lib/node_modules/@stdlib/stats/nanmax/README.md:
```diff
@@ -0,0 +1,273 @@
+
+#### nanmax.assign( x, out\[, options] )
+
+Computes the maximum value along one or more [ndarray][@stdlib/ndarray/ctor] dimensions and assigns results to a provided output [ndarray][@stdlib/ndarray/ctor], ignoring `NaN` values.
```
```suggestion
Computes the maximum value along one or more [ndarray][@stdlib/ndarray/ctor] dimensions while ignoring `NaN` values and assigns results to a provided output [ndarray][@stdlib/ndarray/ctor].
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/nanmax/README.md:
```diff
@@ -0,0 +1,273 @@
+
+#### nanmax.assign( x, out\[, options] )
+
+Computes the maximum value along one or more [ndarray][@stdlib/ndarray/ctor] dimensions while ignoring `NaN` values and assigns results to a provided output [ndarray][@stdlib/ndarray/ctor].
```
```suggestion
Computes the maximum value along one or more [ndarray][@stdlib/ndarray/ctor] dimensions, ignoring `NaN` values, and assigns results to a provided output [ndarray][@stdlib/ndarray/ctor].
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/nanmax/docs/repl.txt:
```diff
@@ -0,0 +1,77 @@
+
+
+{{alias}}.assign( x, out[, options] )
+    Computes the maximum value along one or more ndarray dimensions and assigns
+    results to a provided output ndarray, ignoring `NaN` values.
```
```suggestion
    Computes the maximum value along one or more ndarray dimensions, ignoring
    `NaN` values, and assigns results to a provided output ndarray.
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/nanmax/docs/types/index.d.ts:
```diff
@@ -0,0 +1,151 @@
+	<T = unknown, U = unknown>( x: InputArray<T>, options?: Options ): OutputArray<U>; // NOTE: we lose type specificity here, but retaining specificity would likely be difficult and/or tedious to completely enumerate, as the output ndarray data type is dependent on how `x` interacts with output data type policy and whether that policy has been overridden by `options.dtype`.
+
+	/**
+	* Computes the maximum value along one or more ndarray dimensions and assigns results to a provided output ndarray, ignoring `NaN` values.
```
```suggestion
	* Computes the maximum value along one or more ndarray dimensions, ignoring `NaN` values, and assigns results to a provided output ndarray.
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/nanmax/docs/types/test.ts:
```diff
@@ -0,0 +1,225 @@
+* limitations under the License.
+*/
+
+/* eslint-disable @typescript-eslint/no-unused-expressions, space-in-parens */
```
```suggestion
/* eslint-disable space-in-parens */
```

kgryte committed:
style: re-enable lint rule

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
test: fix failing tests

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM after fixing tests.
